lib/charms/grafana_agent/v0/cos_agent.py

```diff
@@ -700,10 +700,17 @@ def metrics_jobs(self) -> List[Dict]:
 
                 # Apply labels to the scrape jobs
                 for static_config in job.get("static_configs", []):
+                    topo_as_dict = topology.as_dict(excluded_keys=["charm_name"])
                     static_config["labels"] = {
                         # Be sure to keep labels from static_config
                         **static_config.get("labels", {}),
-                        **topology.label_matcher_dict,
+                        # TODO: We should add a new method in juju_topology.py
+                        # that like `as_dict` method, returns the keys with juju_ prefix
```

Comment:
**nitpick**: I'd suggest maybe adding a link to the issue you created as an extra comment line